R/table-subscript.R

```diff
@@ -312,6 +313,45 @@ updateIsSubscriptedAttr <- function(y, x) {
     y
 }
 
+updateCellText <- function(y, x.attributes, evaluated.args) {
+    cell.text <- x.attributes$celltext
+    if (!is.array(cell.text)) {
+        return (y)
+    }
+
+    # convert row/column/stat name args to numeric indices
+    indices <- lapply(seq_along(evaluated.args), function(i) {
+        if (is.character(evaluated.args[[i]]) && !is.null(x.attributes$dimnames)) {
+            dim.names <- x.attributes$dimnames[[i]]
+            all.indices <- seq_along(dim.names)
+            names(all.indices) <- dim.names
+            all.indices[evaluated.args[[i]]]
+        } else {
+            evaluated.args[[i]]
+        }
+    })
+
+    subscripted <- do.call(`[`, c(list(cell.text), indices))
+
+    if (!is.array(subscripted)) {
+        subscripted <- array(subscripted)
+    }
```

Comment:
Is there a need to force it to always be an `array`? Could call `subscripted <- do.call("[", evaluated.args)` and if it gets dropped that is ok? Or does this mess with the way `celltext` is used in the rendering?

Reply:
Same reason as above. It is easier if it is always an array, rather than only sometimes.